pkg/apis/channels/v1alpha1/subscription_types.go

```diff
@@ -57,14 +64,53 @@ type SubscriptionSpec struct {
 	// +optional
 	Generation int64 `json:"generation,omitempty"`
 
-	// Channel is the name of the channel to subscribe to.
-	Channel string `json:"channel"`
-
-	// Subscriber is the name of the subscriber service DNS name.
-	Subscriber string `json:"subscriber"`
-
-	// Target service DNS name for replies returned by the subscriber.
-	ReplyTo string `json:"replyTo,omitempty"`
+	// Reference to an object that will be used to create the subscription
+	// for receiving events. The object must have spec.subscriptions
+	// list which will then be modified accordingly.
+	//
+	// This object must fulfill the Subscribable contract.
+	//
+	// You can specify only the following fields of the ObjectReference:
+	//   - Kind
+	//   - APIVersion
+	//   - Name
+	From *corev1.ObjectReference `json:"from,omitempty"`
+
+	// Processor is reference to (optional) function for processing events.
+	// Events from the From channel will be delivered here and replies
+	// are sent to To channel.
+	//
+	// This object must fulfill the Targetable contract.
+	//
+	// Reference to an object that will be used to deliver events for
+	// (optional) processing before sending them to To for further
+	// if specified for additional Subscriptions to then subscribe
+	// to these events for further processing.
+	//
+	// For example, this could be a reference to a Route resource
+	// or a Service resource.
+	// TODO: Specify the required fields the target object must
+	// have in the status.
+	// You can specify only the following fields of the ObjectReference:
+	//   - Kind
+	//   - APIVersion
+	//   - Name
+	// +optional
+	Processor *corev1.ObjectReference `json:"processor,omitempty"`
+
+	// To is the (optional) resolved channel where (optionally) processed
+	// events get sent.
+	//
+	// This object must fulfill the Sinkable contract.
+	//
+	// TODO: Specify the required fields the target object must
+	// have in the status.
+	// You can specify only the following fields of the ObjectReference:
+	//   - Kind
+	//   - APIVersion
+	//   - Name
```

Comment:
What about UID?

I'm assuming that Namespace is specifically excluded?

Reply:
> What about UID?

Is there a good way to lookup a resource by UID?

> I'm assuming that Namespace is specifically excluded?

Keeping subscriptions within a single namespace is a good idea until we understand fully understand the tenancy consequences of allowing them to cross namespaces.

Reply:
Let's start with names (no UIDs), and a single namespace until we have more understanding on what cases this does not handel?